File: web/backend/Python-services/src/models/Device.py
"""
Device model for FAMS
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from ..constants import COLLECTIONS


class Device(BaseModel):
    """Device model representing hardware devices used in the system"""
    deviceId: int = Field(..., description="Unique identifier for the device")
    deviceName: str = Field(..., description="Name of the device")
    deviceType: str = Field(default="Jetson Nano", description="Type of device")
    location: Optional[str] = Field(None, description="Location of the device")
    status: bool = Field(default=True, description="Whether the device is operational")
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)
    isActive: bool = Field(default=True)
    collection: str = COLLECTIONS['DEVICE'] 

File: web/backend/NodeJS-services/database/models/Device.js
const mongoose = require('mongoose');
const { COLLECTIONS } = require('../constants');

/**
 * Device Schema
 * Represents physical devices used in the system
 */
const DeviceSchema = new mongoose.Schema({
  deviceId: {
    type: Number,
    required: true,
    unique: true,
    auto: true
  },
  deviceName: {
    type: String,
    required: true
  },
  deviceType: {
    type: String,
    default: 'Jetson Nano'
  },
  location: {
    type: String
  },
  status: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true,
  versionKey: false,
  toJSON: { virtuals: true },
  toObject: { virtuals: true }
});

// Virtual for getting models deployed on this device
DeviceSchema.virtual('models', {
  ref: 'ModelVersion',
  localField: 'deviceId',
  foreignField: 'deviceId',
  justOne: false
});

module.exports = mongoose.model('Device', DeviceSchema, COLLECTIONS.DEVICE); 